src/core/backbone_router/bbr_manager.cpp

```diff
@@ -237,16 +239,26 @@ void Manager::HandleMulticastListenerRegistration(const Coap::Message &aMessage,
 
             if (failed)
             {
-                failedAddresses[failedAddressNum++] = address;
+                addresses[failedAddressNum++] = address;
+            }
+            else
+            {
+                // Put successfully registered addresses at the end of `addresses`.
+                addresses[kIPv6AddressesNumMax - (++successAddressNum)] = address;
```

Comment:
Do we need to check the success/failed array indexes to see if we reached the end of array (to ensure we never write beyond end/start of array)?

Also we write failed ones from start and success ones from end of array? Do we need to check the two indexes reach each other (or not possible)?

Reply:
No. It's already guaranteed that we won't overflow `addresses` array and failed ones won't collide with success ones.

The size of `addresses` is `kIPv6AddressesNumMax`. 
We also have this check to make sure that the number of addresses in the message never exceed `kIPv6AddressesNumMax`. So accesses to `addresses` are always valid. 
```c++
    VerifyOrExit(addressesLength / sizeof(Ip6::Address) <= kIPv6AddressesNumMax,
                 status = ThreadStatusTlv::kMlrGeneralFailure);
```

Each address read from message is either `success` or `failed`. So `failedAddressNum + successAddressNum` won't exceed `kIPv6AddressesNumMax` either. So, the two indexes won't reach each other in any case.

Reply:
Sounds good. Thanks.